core/src/test/java/com/linecorp/armeria/client/endpoint/healthcheck/HealthCheckedEndpointGroupTest.java

```diff
@@ -34,6 +38,43 @@
 import com.linecorp.armeria.common.util.AsyncCloseable;
 
 class HealthCheckedEndpointGroupTest {
+
+    @Test
+    void delegateUpdateCandidatesWhileCreatingHealthCheckedEndpointGroup() {
+        final MockEndpointGroup delegate = new MockEndpointGroup();
+        final CompletableFuture<List<Endpoint>> future = delegate.initialEndpointsFuture();
+        future.complete(ImmutableList.of(Endpoint.of("127.0.0.1", 8080), Endpoint.of("127.0.0.1", 8081)));
+
+        final CountDownLatch latch = new CountDownLatch(1);
+
+        // Schedule the task which update the endpoint one second later to ensure that the change is happening
+        // while creating the HealthCheckedEndpointGroup.
+        Executors.newSingleThreadScheduledExecutor().schedule(
```

Comment:
nit: use daemon thread?
`Executors.newSingleThreadScheduledExecutor(new DefaultThreadFactory("health-check", true))`

Reply:
Fixed to use `EventLoopExtension`. 😉

Reply:
Can also use a common event loop or forkjoinpool

Reply:
Changed to use `CommonPools`. (Can't use forkjoinpool because I have to schedule it. 😉 )